editoast/src/views/v2/train_schedule/projection.rs

```diff
@@ -0,0 +1,599 @@
+            .await?;
+        if let Some(cached) = projection {
+            hit_cache.insert(*train_id, cached);
+        } else {
+            miss_cache.insert(*train_id, train_details);
+        }
+    }
+    if miss_cache.is_empty() {
+        info!("Hit cache");
+    }
+
+    // 2 Compute space time curves and signal updates for all miss cache
+    let (space_time_curves, signal_updates) = join!(
+        compute_batch_space_time_curves(&miss_cache, &path_projection),
+        compute_batch_signal_updates(core_client.clone(), &miss_cache)
+    );
+
+    // 3. Store the projection in the cache
+    for (id, train_details) in miss_cache {
+        let hash = train_projection_input_hash(
+            infra.id,
+            &infra.version,
+            &train_details,
+            &path_track_ranges,
+        );
+        let cached = CachedProjectPathTrainResult {
+            space_time_curves: space_time_curves
+                .get(&id)
+                .expect("Space time curves not availabe for train")
+                .clone(),
+            signal_updates: signal_updates
+                .get(&id)
+                .expect("Signal update not availabe for train {}")
+                .clone(),
+        };
+        redis_conn
+            .json_set_ex(&hash, &cached, CACHE_PROJECTION_EXPIRATION)
+            .await?;
+        hit_cache.insert(id, cached);
+    }
+
+    // 4.1 Fetch rolling stock length
+    let mut project_path_result = HashMap::new();
+    let rolling_stock_names: HashSet<_> = hit_cache
+        .keys()
+        .map(|id| train_map.get(id).unwrap().rolling_stock_name.clone())
+        .collect();
+    let (rs, missing): (Vec<_>, _) =
+        RollingStockModel::retrieve_batch(conn, rolling_stock_names).await?;
+    assert!(missing.is_empty(), "Missing rolling stock models");
+    let rolling_stock_length: HashMap<_, _> =
+        rs.into_iter().map(|rs| (rs.name, rs.length)).collect();
+
+    // 4.2 Build the projection response
+    for (id, cached) in hit_cache {
+        let train = train_map.get(&id).expect("Train not found");
+        let length = rolling_stock_length
+            .get(&train.rolling_stock_name)
+            .expect("Rolling stock length not found");
+
+        project_path_result.insert(
+            id,
+            ProjectPathTrainResult {
+                departure_time: train.start_time,
+                rolling_stock_length: (length * 1000.).round() as u64,
+                cached,
+            },
+        );
+    }
+
+    Ok(Json(project_path_result))
+}
+
+/// Input for the projection of a train schedule on a path
+#[derive(Debug, Clone, Hash)]
+struct TrainSimulationDetails {
+    positions: Vec<u64>,
+    times: Vec<u64>,
+    train_path: Vec<TrackRange>,
+    signal_sightings: Vec<SignalSighting>,
+    zone_updates: Vec<ZoneUpdate>,
+}
+
+/// Compute the signal updates of a list of train schedules
+async fn compute_batch_signal_updates(
+    _core_client: Data<CoreClient>,
```

Comment:
```suggestion
    _core_client: Arc<CoreClient>,
```

I think we should try to keep the `Data<T>` only in endpoint functions, otherwise we might start leaking actix everywhere like we did for the DbPool